### ISSUE TITLE
Add negative color/prefix pattern condition

### ISSUE THREAD

**Issue by J0WI** [coursehero/slacktee#58]
It would be useful to have negative conditions. E.g. to consider all lines that do not start with "Success" as failure:
```
tail -f app.log | slacktee.sh -n -a "good" -no "danger" "^Success:" -nd "@channel" "^Success:" -m link_names
```

**Comment by Chikashi-Kato** [comment]
Hello, thank you for submitting the enhancement idea. I think it's useful for the case which uses complicated regex pattern. For the simple case (like your example), we can use negated character class. 

```
./slacktee.sh -c "chuck_test" -n -a "good" -o "danger" "^[^(Success)]" -d "@channel" "^[^(Success)]" -m link_names
```
I am not 100% sure if `^[^(Success)]` is a correct regex pattern, but it works like a charm as far as I hav e tested ;p

**Comment by Hoten** [comment]
> ^[^(Success)]

Unfortunately that "negated" regex works by accident. Any character in brackets are interpreted directly, except for dashes for denoting ranges and an anchor to negate the character class, AFAIK. So group capturing does not work. [See this](https://regex101.com/r/a3D2m2/1).

A [negative lookaround](https://regex101.com/r/mduA7I/1) will work.

```
^(?!Success)
```

Anyways, a negate option would be useful! Thank you for the suggestion. Would you like to give a shot at an implementation?

**Comment by Chikashi-Kato** [comment]
Hi Connor, thank you for checking it! Yeah, I checked `^[^(Success)]` with multiple regex parser, but none of them worked like Bash's regex operator (`=~`) which slacktee is using.

I also tried to use negative lookaround, but it didn't work with Bash's regex operator. Seems like the implementation of Bash's regex operator is kinda unique ;p

I would love to have the negate option, but I am not sure I can quickly find enough time to implement it by myself (it should be super easy to implement, though). It would be really appreciated if anyone can implement it and submit PR!

**Comment by J0WI** [comment]
How about negating the condition, like `if ! [[ $line =~ ${cond_prefix_patterns[$i]} ]]`?

**Comment by Chikashi-Kato** [comment]
Yeah, it makes sense :) We should flip the result of the evaluation if the negate option is specified :)
